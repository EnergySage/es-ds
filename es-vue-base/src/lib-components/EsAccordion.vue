<template>
    <div class="EsAccordion border-bottom border-light rounded-bottom">
        <header
            role="tab"
            class="position-relative">
            <component
                :is="headingTag"
                class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between px-100 px-sm-200 py-100 text-body font-size-100"
                :class="{
                    'bg-gray-200': isVisible,
                    'bg-white': !isVisible,
                    'EsAccordion-heading--visible': isVisible,
                }">
                <slot name="title" />
                <icon-chevron-down class="EsAccordion-icon flex-shrink-0 ml-200" />
            </component>
            <button
                class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"
                @click="handleClick">
                <span class="sr-only">{{ isVisible ? 'collapse' : 'expand' }}</span>
            </button>
        </header>
        <b-collapse
            :id="id"
            :visible="isVisible"
            role="tabpanel">
            <div class="EsAccordion-content bg-white px-100 px-sm-200 pb-25 pt-100">
                <slot />
            </div>
        </b-collapse>
    </div>
</template>

<script>
import { BCollapse } from 'bootstrap-vue';
import IconChevronDown from '../lib-icons/icon-chevron-down.vue';

export default {
    name: 'EsAccordion',
    components: {
        BCollapse,
        IconChevronDown,
    },
    /**
     * Receive the injected props that EsAccordionList provides.
     */
    inject: {
        /**
         * A array of ids for accordions that are expanded.
         */
        expandedIds: {
            type: Array,
            required: true,
        },
        /**
         * A reference to the parent EsAccordionList so we can emit events onto it directly.
         */
        parent: {
            type: Object,
            required: true,
        },
    },
    props: {
        /**
         * A unique id used to keep track of which accordion is expanded.
         */
        id: {
            type: String,
            required: true,
        },
        headingTag: {
            type: String,
            default: 'h3',
        },
    },
    computed: {
        isVisible() {
            return this.expandedIds.includes(this.id);
        },
    },
    methods: {
        /**
         * Tell the parent EsAccordionList component (which keeps track of which
         * accordion is expanded) that this one wants to expand or collapse.
         * @event accordion-toggle
         */
        handleClick() {
            this.parent.$emit('accordion-toggle', this.id);
        },
    },
};
</script>

<style lang="scss" scoped>
@import '~@energysage/es-bs-base/scss/includes';

/* rotate the chevron when expanded */
.EsAccordion-heading--visible {
    .EsAccordion-icon {
        transform: rotate(180deg);
    }
}

/* for smooth background color transition */
.EsAccordion-heading {
    transition: $transition-base;
}

/* first item needs rounded corners on top */
.EsAccordion:first-child {
    .EsAccordion-heading {
        border-radius: $border-radius $border-radius 0 0 !important;
    }
}
/**
 * last item needs rounded corners on bottom of the button or content,
 * depending which is visible at the bottom of the list
 */
.EsAccordion:last-child {
    .EsAccordion-heading,
    .EsAccordion-content {
        border-radius: 0 0 $border-radius $border-radius !important;
    }
    /* if the last item is expanded, the button is no longer the last visible item */
    .EsAccordion-heading--visible {
        border-radius: 0 !important;
    }
}
/* align button to cover the heading tag */
.EsAccordion-button {
    top: 0;
}

/* only animate the chevron if the user doesn't prefer reduced motion */
@media not prefers-reduced-motion {
    .EsAccordion-icon {
        transition: $transition-base;
    }
}
</style>
