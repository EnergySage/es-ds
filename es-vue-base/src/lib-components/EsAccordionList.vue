<template>
    <div
        class="rounded"
        role="tablist">
        <slot />
    </div>
</template>

<script>
import { computed } from 'vue';

export default {
    name: 'EsAccordionList',
    /**
     * Provide the below items to children EsAccordion components that listen for them via inject.
     * Make expandedIds reactive as per https://vuejs.org/guide/components/provide-inject.html#working-with-reactivity.
     */
    provide() {
        return {
            expandedIds: computed(() => this.expandedIds),
            parent: this,
            variant: this.variant,
        };
    },
    props: {
    /**
     * True if we should allow multiple items to be expanded simultaneously,
     * false if we should only allow one (default accordion behavior).
     */
        allowMultipleExpand: {
            type: Boolean,
            default: false,
        },
        /**
         * The id of the accordion that should be expanded initially, if any.
         * Do not use at the same time as v-model.
         */
        initialExpandedId: {
            type: String,
            default: '',
        },
        /**
         * If v-model is used, this will contain a two-way binding to the id of the accordion that is expanded, if any.
         */
        value: {
            type: String,
            default: '',
        },
        /**
         * Used for styling purposes. At this time only two options are allowed: default and minimal.
         */
        variant: {
            type: String,
            validator: (value) => ['default', 'minimal'].includes(value),
            default: 'default',
        },
    },
    data() {
        return {
            // will take either prop as initial value, but value (from v-model) will supersede if present
            expandedIds: (this.value || this.initialExpandedId) ? [this.value || this.initialExpandedId] : [],
        };
    },
    watch: {
        /**
         * If the v-model value changes, update the internal state.
         */
        value(newValue) {
            this.toggleId(newValue);
        },
        /**
         * If the list of expanded ids changes, fire the input event in case v-model is present.
         * Assume only one id can be expanded at a time.
         * @emits input
         */
        expandedIds(newValue) {
            this.$emit('input', newValue.length ? newValue[0] : '');
        },
    },
    /**
     * Listen for @accordion-toggle event from children EsAccordion components.
     * They will inject the 'parent' object this component provides and call this.parent.$emit('accordion-toggle').
     * @emits input
     */
    created() {
        this.$on('accordion-toggle', (id) => {
            if (this.value) {
                // if we're using v-model, fire the input event and
                // have the change come via the value watch function
                this.$emit('input', id);
            } else {
                // if we're not using v-model, toggle the id directly
                this.toggleId(id);
            }
        });
    },
    methods: {
        toggleId(id) {
            if (!id) return;

            const indexOfId = this.expandedIds.indexOf(id);
            if (indexOfId > -1) {
                // if id is expanded, remove it from array
                this.expandedIds.splice(indexOfId, 1);
            } else {
                // if id is not expanded, add it to the array
                // if we don't allow multiple expand, make sure it's the only item in the new array
                this.expandedIds = this.allowMultipleExpand
                    ? [...this.expandedIds, id]
                    : [id];
            }
        },
    },
};
</script>
