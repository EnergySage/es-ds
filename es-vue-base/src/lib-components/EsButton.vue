<template>
    <b-button
        :class="{ 'inline': inline }"
        :variant="computedVariant"
        :size="size"
        v-bind="$attrs"
        v-on="$listeners">
        <!--
            @slot Button Content
            @binding {string} text or html of the button content
        -->
        <slot />
    </b-button>
</template>

<script lang="js">
import { BButton } from 'bootstrap-vue';

export default {
    name: 'EsButton',
    components: {
        BButton,
    },
    props: {
        /**
         * Button Variant
         *
         */
        variant: {
            type: String,
            default: 'primary',
            required: false,
        },
        /**
         * Outline only button
         */
        outline: {
            type: Boolean,
            default: false,
        },
        /**
         * Size of button
         */
        size: {
            type: String,
            validator: (val) => ['lg', 'md', 'sm'].includes(val),
            default: 'md',
        },
        /**
         * Only works for 'link' variant buttons.
         * Removes the fixed height and padding so the button can fit nicely within a block of text.
         */
        inline: {
            type: Boolean,
            default: false,
        },
    },
    computed: {
        /**
         * Compute the Variant
         * used for when you want to `outline` something but don't know the
         * theme's default variant
         *
         * @returns { string } computed variant based on outline prop
         */
        computedVariant() {
            return `${this.outline && this.variant !== 'link' ? 'outline-' : ''}${this.variant}`;
        },
    },
};
</script>

<style lang="scss" scoped>
.btn-link.inline {
    /* use normal CSS here so users can override with utility classes as necessary */
    border: 0;
    height: auto;
    line-height: inherit;
    margin: 0;
    padding: 0;
    vertical-align: baseline;
}
</style>
