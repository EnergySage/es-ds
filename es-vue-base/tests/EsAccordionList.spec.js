import { mount } from '@vue/test-utils';
import EsAccordionList from '@/src/lib-components/EsAccordionList.vue';
import EsAccordion from '@/src/lib-components/EsAccordion.vue';
import EsButton from '@/src/lib-components/EsButton.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsAccordion', () => {
    // Basic test; should exist for most components
    test('<EsAccordionList />', async () => {
        const wrapper = mount(EsAccordionList, {
            ...jestVue,
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();

        // verify prop values are their default values
        expect(wrapper.props('allowMultipleExpand')).toBe(false);
        expect(wrapper.props('initialExpandedId')).toBe('');
    });

    test('EsAccordionList mounts with initial expand; collapse also works', async () => {
        const id = 'accordion-id';
        const title = 'My Title';
        const content = 'My content.';
        const wrapper = mount(EsAccordionList, {
            ...jestVue,
            propsData: {
                initialExpandedId: id,
            },
            slots: {
                default: `
                    <es-accordion id="${id}">
                        <template #title>${title}</template>
                        <p>${content}</p>
                    </es-accordion>
                `,
            },
            stubs: {
                'es-accordion': EsAccordion,
            },
        });

        // verify prop value is correct
        expect(wrapper.props('initialExpandedId')).toBe(id);

        // verify snapshot is correct
        expect(wrapper.html()).toMatchSnapshot();

        // verify title slot
        const button = wrapper.findComponent(EsButton);
        expect(button.text()).toContain(title);

        // verify title and content slots together
        expect(wrapper.text()).toBe(`${title}   ${content}`);

        // verify that the content is visible
        const paragraph = wrapper.find('p');
        expect(paragraph.exists()).toBe(true);
        expect(paragraph.isVisible()).toBe(true);

        // collapse the accordion
        await button.trigger('click');

        // verify that the content is not visible anymore
        expect(paragraph.isVisible()).toBe(false);
    });

    test('EsAccordionList expands one item and collapses another', async () => {
        const accordions = [
            {
                id: 'accordion-id-1',
                title: 'My Title 1',
                content: 'My content 1.',
            },
            {
                id: 'accordion-id-2',
                title: 'My Title 2',
                content: 'My content 2.',
            },
        ];

        const wrapper = mount(EsAccordionList, {
            ...jestVue,
            propsData: {
                initialExpandedId: accordions[1].id,
            },
            slots: {
                default: accordions.map((accordion) => `
                    <es-accordion id="${accordion.id}">
                        <template #title>${accordion.title}</template>
                        <p>${accordion.content}</p>
                    </es-accordion>
                `).join(''),
            },
            stubs: {
                'es-accordion': EsAccordion,
            },
        });

        // verify prop value is correct
        expect(wrapper.props('initialExpandedId')).toBe(accordions[1].id);

        // verify snapshot is correct
        expect(wrapper.html()).toMatchSnapshot();

        // verify the first item is not expanded
        const firstItem = wrapper.find(`#${accordions[0].id}`);
        expect(firstItem.exists()).toBe(true);
        const firstItemContent = firstItem.find('p');
        expect(firstItemContent.exists()).toBe(true);
        expect(firstItemContent.isVisible()).toBe(false);

        // verify the second item is expanded
        const secondItem = wrapper.find(`#${accordions[1].id}`);
        expect(secondItem.exists()).toBe(true);
        const secondItemContent = secondItem.find('p');
        expect(secondItemContent.exists()).toBe(true);
        expect(secondItemContent.isVisible()).toBe(true);

        // find and click the first item's button to expand it
        const firstAccordion = wrapper.findAllComponents(EsAccordion).at(0);
        expect(firstAccordion.exists()).toBe(true);
        const firstButton = firstAccordion.findComponent(EsButton);
        expect(firstButton.exists()).toBe(true);
        await firstButton.trigger('click');

        // verify the first item is expanded
        expect(firstItemContent.isVisible()).toBe(true);

        // verify the second item is not expanded
        expect(secondItemContent.isVisible()).toBe(false);
    });

    test('EsAccordionList performs multi-expand correctly', async () => {
        const accordions = [
            {
                id: 'accordion-id-1',
                title: 'My Title 1',
                content: 'My content 1.',
            },
            {
                id: 'accordion-id-2',
                title: 'My Title 2',
                content: 'My content 2.',
            },
        ];

        const wrapper = mount(EsAccordionList, {
            ...jestVue,
            propsData: {
                allowMultipleExpand: true,
                initialExpandedId: accordions[0].id,
            },
            slots: {
                default: accordions.map((accordion) => `
                    <es-accordion id="${accordion.id}">
                        <template #title>${accordion.title}</template>
                        <p>${accordion.content}</p>
                    </es-accordion>
                `).join(''),
            },
            stubs: {
                'es-accordion': EsAccordion,
            },
        });

        // verify prop values are correct
        expect(wrapper.props('allowMultipleExpand')).toBe(true);
        expect(wrapper.props('initialExpandedId')).toBe(accordions[0].id);

        // verify snapshot is correct
        expect(wrapper.html()).toMatchSnapshot();

        // verify the first item is expanded
        const firstItem = wrapper.find(`#${accordions[0].id}`);
        expect(firstItem.exists()).toBe(true);
        const firstItemContent = firstItem.find('p');
        expect(firstItemContent.exists()).toBe(true);
        expect(firstItemContent.isVisible()).toBe(true);

        // verify the second item is not expanded
        const secondItem = wrapper.find(`#${accordions[1].id}`);
        expect(secondItem.exists()).toBe(true);
        const secondItemContent = secondItem.find('p');
        expect(secondItemContent.exists()).toBe(true);
        expect(secondItemContent.isVisible()).toBe(false);

        // find and click the second item's button to expand it
        const secondAccordion = wrapper.findAllComponents(EsAccordion).at(1);
        expect(secondAccordion.exists()).toBe(true);
        const secondButton = secondAccordion.findComponent(EsButton);
        expect(secondButton.exists()).toBe(true);
        await secondButton.trigger('click');

        // verify the first item is expanded
        expect(firstItemContent.isVisible()).toBe(true);

        // verify the second item is also expanded
        expect(secondItemContent.isVisible()).toBe(true);
    });
});
