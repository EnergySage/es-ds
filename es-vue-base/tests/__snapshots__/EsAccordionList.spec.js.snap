// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsAccordion <EsAccordionList /> 1`] = `"<div role="tablist" class="rounded"></div>"`;

exports[`EsAccordion EsAccordionList expands one item and collapses another 1`] = `
"<div role="tablist" class="rounded">
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-white font-size-100 px-100 px-sm-200">My Title 1 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">expand</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id-1" class="collapse" style="display: none;">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content 1.</p>
        </div>
      </div>
    </transition-stub>
  </div>
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-gray-200 EsAccordion-heading--visible font-size-100 px-100 px-sm-200">My Title 2 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">collapse</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id-2" class="collapse show">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content 2.</p>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`EsAccordion EsAccordionList mounts with initial expand; collapse also works 1`] = `
"<div role="tablist" class="rounded">
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-gray-200 EsAccordion-heading--visible font-size-100 px-100 px-sm-200">My Title <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">collapse</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id" class="collapse show">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content.</p>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`EsAccordion EsAccordionList performs multi-expand correctly 1`] = `
"<div role="tablist" class="rounded">
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-gray-200 EsAccordion-heading--visible font-size-100 px-100 px-sm-200">My Title 1 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">collapse</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id-1" class="collapse show">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content 1.</p>
        </div>
      </div>
    </transition-stub>
  </div>
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-white font-size-100 px-100 px-sm-200">My Title 2 <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">expand</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id-2" class="collapse" style="display: none;">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content 2.</p>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`EsAccordion EsAccordionList v-model test 1`] = `
"<div role="tablist" class="rounded" data-testid="testInput" id="testId">
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-gray-200 EsAccordion-heading--visible font-size-100 px-100 px-sm-200">
        My Title 1
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">collapse</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id-1" class="collapse show">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content 1.</p>
        </div>
      </div>
    </transition-stub>
  </div>
  <div role="tab" class="EsAccordion border-bottom border-light EsAccordion--default rounded-bottom">
    <header class="position-relative">
      <h3 class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body bg-white font-size-100 px-100 px-sm-200">
        My Title 2
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" style="height: 24px; width: 24px;" class="EsAccordion-icon flex-shrink-0 ml-200">
          <path fill-rule="evenodd" clip-rule="evenodd" d="M3.293 8.293a1 1 0 0 1 1.414 0L12 15.586l7.293-7.293a1 1 0 1 1 1.414 1.414l-8 8a1 1 0 0 1-1.414 0l-8-8a1 1 0 0 1 0-1.414Z"></path>
        </svg></h3> <button class="EsAccordion-button position-absolute w-100 h-100 border-0 bg-transparent"><span class="sr-only">expand</span></button>
    </header>
    <transition-stub css="true" enterclass="" leaveclass="collapse show" entertoclass="collapse show" leavetoclass="collapse" enteractiveclass="collapsing" leaveactiveclass="collapsing" role="tabpanel">
      <div id="accordion-id-2" class="collapse" style="display: none;">
        <div class="EsAccordion-content pb-25 bg-white pt-100 px-100 px-sm-200">
          <p>My content 2.</p>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;
